ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerSelectionActions.hs

```diff
@@ -96,35 +115,56 @@ withPeerSelectionActions
   readPeerSharingController
   readNewInboundConnections
   peerStateActions
-  getLedgerPeers
+  ledgerPeersRng
+  ledgerPeersConsensusInterface
+  getUseLedgerAfter
   k = do
     localRootsVar <- newTVarIO mempty
+    reqVar  <- newEmptyTMVarIO
+    respVar <- newEmptyTMVarIO
+    let getRequest  = takeTMVar reqVar
+        putResponse = putTMVar  respVar
+        getLedgerPeers :: NumberOfPeers -> LedgerPeersKind -> m (Maybe (Set peeraddr, DiffTime))
+        getLedgerPeers = \numberOfPeers ledgerPeersKind -> do
+          atomically $ putTMVar reqVar (numberOfPeers, ledgerPeersKind)
+          atomically $ takeTMVar respVar
```

Comment:
This should be hidden, these should be internal details of the ledger peers provider.